modules/ting_das/includes/ting_das.client.inc

```diff
@@ -0,0 +1,430 @@
+  private $publicationDateOfComponent = NULL;
+  private $publicationYearOfComponent = NULL;
+  private $issueOfComponent = NULL;
+  private $volumeOfComponent = NULL;
+  private $pagesOfComponent = NULL;
+
+  /**
+   * TingDasCopyRequest constructor.
+   *
+   * @param string $url
+   *   The service endpoint for digital article service.
+   * @param string $username
+   *   The username to login with.
+   * @param string $password
+   *   The password to login with.
+   */
+  public function __construct(string $url, string $username, string $password) {
+    $this->url = $url;
+    $this->wsUsername = $username;
+    $this->wsPassword = $password;
+  }
+
+  /**
+   * Set date well id.
+   *
+   * @param string $pid
+   *   The date well id for the material requested.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPid(string $pid) {
+    $this->pid = $pid;
+    return $this;
+  }
+
+  /**
+   * The data well id.
+   *
+   * @return string|null
+   *   The data well pid if set else NULL.
+   */
+  public function getPid() {
+    return $this->pid;
+  }
+
+  /**
+   * Set the users loaner id.
+   *
+   * @param string $userLoanerId
+   *   The users loaner id (borrow id/CPR).
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserLoanerId(string $userLoanerId) {
+    $this->userLoanerId = $userLoanerId;
+    return $this;
+  }
+
+  /**
+   * Get the user loaner id.
+   *
+   * @return string|null
+   *   The loaner id or null if not set.
+   */
+  public function getUserLoanerId() {
+    return $this->userLoanerId;
+  }
+
+  /**
+   * Set the username (full name of the user).
+   *
+   * @param string $userName
+   *   The users full name.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserName(string $userName) {
+    $this->userName = $userName;
+    return $this;
+  }
+
+  /**
+   * Get the users full name.
+   *
+   * @return string|null
+   *   The users full name.
+   */
+  public function getUserName() {
+    return $this->userName;
+  }
+
+  /**
+   * Set the users e-mail-address.
+   *
+   * @param string $mail
+   *   The users e-mail-address.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserMail(string $mail) {
+    $this->mail = $mail;
+    return $this;
+  }
+
+  /**
+   * Get the users e-mail-address.
+   *
+   * @return string|null
+   *   The users e-mail-address.
+   */
+  public function getUserMail() {
+    return $this->mail;
+  }
+
+  /**
+   * The interest date for the user.
+   *
+   * The date until the user is not longer interested in getting the material.
+   *
+   * @param string $userInterestDays
+   *   The users interest date in number of days.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserInterestDate(string $userInterestDays) {
```

Comment:
If this accepts a number of days then `int` seems more appropriate than `string`?

Reply:
Changed.